module-system/module-implementations/sov-attester-incentives/src/call.rs

```diff
@@ -0,0 +1,764 @@
+                .light_client_finalized_height
+                .get(working_set)
+                .expect("Must be set at genesis");
+
+            // Remove the attester from the bonding set
+            self.bonded_attesters.remove(context.sender(), working_set);
+
+            // Then add the bonded attester to the unbonding set, with the current finalized height
+            self.unbonding_attesters.set(
+                context.sender(),
+                &UnbondingInfo {
+                    unbonding_initiated_height: finalized_height,
+                    amount: bond,
+                },
+                working_set,
+            );
+        }
+
+        Ok(CallResponse::default())
+    }
+
+    pub(crate) fn end_unbond_attester(
+        &self,
+        context: &C,
+        working_set: &mut WorkingSet<C::Storage>,
+    ) -> Result<sov_modules_api::CallResponse, AttesterIncentiveErrors> {
+        // We have to ensure that the attester is unbonding, and that the unbonding transaction
+        // occurred at least `finality_period` blocks ago to let the attester unbond
+        if let Some(unbonding_info) = self.unbonding_attesters.get(context.sender(), working_set) {
+            // These two constants should always be set beforehand, hence we can panic if they're not set
+            let curr_height = self
+                .light_client_finalized_height
+                .get(working_set)
+                .expect("Should be defined at genesis");
+            let finality_period = self
+                .rollup_finality_period
+                .get(working_set)
+                .expect("Should be defined at genesis");
+
+            if unbonding_info.unbonding_initiated_height + finality_period > curr_height {
+                return Err(self.slash_burn_reward(
+                    context.sender(),
+                    Role::Attester,
+                    SlashingReason::UnbondingNotFinalized,
+                    working_set,
+                ));
+            }
+
+            // Get the user's old balance.
+            // Transfer the bond amount from the sender to the module's address.
+            // On failure, no state is changed
+            self.reward_sender(context, unbonding_info.amount, working_set)?;
+
+            // Update our internal tracking of the total bonded amount for the sender.
+            self.bonded_attesters.remove(context.sender(), working_set);
+            self.unbonding_attesters
+                .remove(context.sender(), working_set);
+
+            // Emit the unbonding event
+            working_set.add_event("unbonded_challenger", {
+                let amount = unbonding_info.amount;
+                &format!("amount_withdrawn: {:?}", amount)
+            });
+        } else {
+            return Err(self.slash_burn_reward(
+                context.sender(),
+                Role::Attester,
+                SlashingReason::AttesterIsNotUnbonding,
+                working_set,
+            ));
```

Comment:
Same here - I don't think that this needs to be slashed

Reply:
Fixed